File: src/main/java/edu/metrostate/trackingsystem/domain/repositories/VehicleRepository.java
package edu.metrostate.trackingsystem.domain.repositories;

import edu.metrostate.trackingsystem.domain.models.Vehicle;
import edu.metrostate.trackingsystem.infrastructure.database.IDatabaseContext;
import edu.metrostate.trackingsystem.infrastructure.database.Result;
import javafx.collections.ObservableList;

import java.util.List;

public class VehicleRepository {

    private final IDatabaseContext databaseContext;

    public VehicleRepository(IDatabaseContext databaseContext) {
        this.databaseContext = databaseContext;
    }

    public Result<Boolean> deleteVehicle(String id, String dealerId) {
        return databaseContext.deleteVehicle(id, dealerId);
    }

    public Result<Boolean> addVehicle(Vehicle vehicle) {
        return databaseContext.addVehicle(vehicle);
    }

    public ObservableList<Vehicle> getVehicles() {
        return databaseContext.getVehicles();
    }

    public void toggleIsRented(Vehicle vehicle) { databaseContext.toggleIsRented(vehicle);}
}

File: src/main/kotlin/edu/metrostate/dealership/domain/repositories/VehicleRepository.kt
//Used for transferring data between application layers from the database (domain layer) to the services layer
package edu.metrostate.dealership.domain.repositories

import edu.metrostate.dealership.domain.models.Vehicle
import edu.metrostate.dealership.infrastructure.database.Database
import edu.metrostate.dealership.infrastructure.database.Result
import javafx.collections.ObservableList

class VehicleRepository(private val databaseContext: Database) {
    fun deleteVehicle(id: String) {
        databaseContext.deleteVehicle(id)
    }

    fun addVehicle(vehicle: Vehicle?): Result<Boolean>? {
        return vehicle?.let { databaseContext.addVehicle(it) }
    }

    val vehicles: ObservableList<Vehicle>
        get() = databaseContext.vehicles

    fun toggleIsRented(vehicleId: String) {
            databaseContext.toggleIsRented(vehicleId)
    }
}